lib/driver-cli.js

```diff
@@ -0,0 +1,247 @@
+                      'it should be of the form <org>/<repo>');
+    }
+    driverData = await installViaNpm({appiumHome, json, driver: installSpec});
+  } else if (installType === INSTALL_TYPE_GIT) {
+    driverData = await installViaNpm({appiumHome, json, driver: installSpec});
+  } else {
+    // at this point we have either an npm package or an appium verified driver
+    // name. both of which will be installed via npm.
+    // drivers installed via npm can include versions or tags after the '@'
+    // sign, so check for that
+    const [name, pkgVer] = installSpec.split('@');
+    let pkgName;
+
+    if (installType === INSTALL_TYPE_NPM) {
+      // if we're installing a named package from npm, we don't need to check
+      // against the appium driver list; just use the installSpec as is
+      pkgName = name;
+    } else {
+      // if we're installing a named appium driver (like 'xcuitest') we need to
+      // dereference the actual npm package ('appiupm-xcuitest-driver'), so
+      // check it exists and get the correct package
+      const knownNames = Object.keys(KNOWN_DRIVERS);
+      if (!_.includes(knownNames, name)) {
+        const msg = `Could not resolve driver; are you sure it's in the list ` +
+                    `of supported drivers? ${JSON.stringify(knownNames)}`;
+        throw new Error(msg);
+      }
+      pkgName = KNOWN_DRIVERS[name];
+      // given that we'll use the install type in the driver json, store it as
+      // 'npm' now
+      installType = INSTALL_TYPE_NPM;
+    }
+
+    driverData = await installViaNpm({appiumHome, json, driver, pkgName, pkgVer});
+  }
+
+  const {driverName} = driverData;
+  delete driverData.driverName;
+
+  if (_.includes(Object.keys(INSTALLED_DRIVERS), driverName)) {
+    throw new Error(`A driver named '${driverName}' is already installed. ` +
+                    `Did you mean to update? 'appium driver update'. See ` +
+                    `installed drivers with 'appium driver list --installed'.`);
+  }
+
+  driverData.installType = installType;
+  driverData.installSpec = installSpec;
+  await writeInstalledDriver(appiumHome, driverName, driverData);
+
+  // refresh the installed driver manifest
+  INSTALLED_DRIVERS = await validateDriversYaml(appiumHome, (msg) => log(json, msg));
+
+  // log info for the user
+  log(json, `Driver ${driverName}@${driverData.version} successfully installed`.green);
+  log(json, `- automationName: ${driverData.automationName.green}`);
+  log(json, `- platformNames: ${JSON.stringify(driverData.platformNames).green}`);
+
+  return INSTALLED_DRIVERS;
+}
+
+async function installViaNpm ({appiumHome, json, driver, pkgName, pkgVer}) {
+  if (!pkgName) {
+    pkgName = driver;
+  }
+  const npmSpec = `${pkgName}${pkgVer ? '@' + pkgVer : ''}`;
+  const specMsg = npmSpec === driver ? '' : ` using NPM install spec '${npmSpec}'`;
+  const msg = `Installing '${driver}'${specMsg}`;
+  try {
+    const pkgJsonData = await spinWith(json, msg, async () => {
```

Comment:
`await spinWith(json, msg, () => installPackage(appiumHome, pkgName, pkgVer))`

Reply:
Was just too long for one line.